hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -166,153 +163,58 @@ protected void doRefresh() {
     refreshFromMetadataLocation(metadataLocation, metadataRefreshMaxRetries);
   }
 
-  @SuppressWarnings("checkstyle:CyclomaticComplexity")
   @Override
   protected void doCommit(TableMetadata base, TableMetadata metadata) {
     boolean newTable = base == null;
     String newMetadataLocation = writeNewMetadataIfRequired(newTable, metadata);
-    boolean hiveEngineEnabled = hiveEngineEnabled(metadata, conf);
-    boolean keepHiveStats = conf.getBoolean(ConfigProperties.KEEP_HIVE_STATS, false);
-
-    CommitStatus commitStatus = CommitStatus.FAILURE;
-    boolean updateHiveTable = false;
-
     HiveLock lock = lockObject(metadata);
```

Comment:
Is it possible to put 
```
try {
      HiveLock lock = lockObject();
      ...
 finally {
     lock.unlock()
 }
```

in the common method as well?

Reply:
[HiveLock](https://github.com/apache/iceberg/blob/4446e4f0abe5c040712ea103594ec31ad8ce902e/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveLock.java)  is not exposed as public. 

I though to keep lock and unlock operations with doCommit. Else I can move only unlock part with `commitWithLocking` . 
WDYT ?